spectec/src/il/validation.ml

```diff
@@ -526,7 +526,7 @@ and valid_prem env prem =
     valid_expmix env mixop e (find "relation" env.rels id) e.at
   | IfPr e ->
     valid_exp env e (BoolT $ e.at)
-  | LetPr (e1, e2) ->
+  | LetPr (e1, e2, _ids) ->
```

Comment:
What is the semantics of these ids? Shouldn't they be checked here?

Reply:
`ids` are there for partial bindings.

For example, if the premise `-- where (a, b) = c`, is a binding for `a`, but not for `b`, then the `ids` will be the list `["a"]`.

I added a check if `ids` is a subset of the free variables in the left-hand side expression `e1`.